bin/flash.flashair

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2016 Konsulko Group. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+# Make sure that the card is available to us now.
+RC=`ping -c 1 ${flashair_ip} >/dev/null 2>&1; echo $?`
```

Comment:
You could avoid duplicating the command:

while true; do
    ping ...
    if [ $? -eq 0 ]; then
        break
    fi
end

Also, shouldn't this have a timeout? I suppose Jenkins (if you're using that) can impose one itself, but any timeout it imposes would have to be large to account for overall test runtime. Perhaps:

i=0
while true; do
    ping ...
    if [ $? -eq 0 ]; then
        break
    fi
    i=$((i + 1))
    if [ $i -eq 30 ]; then # or 60 or whatever seems reasonable in practice
        echo FlashAir failed to ping within timeout
        exit 1
    fi
    sleep 1
end

Also, is ping a reliable enough test, or do we need to httping to make sure the web server is up not just the IP stack? Perhaps the device is "embedded" enough the two are basically equivalent?